automata/base/automaton.py

```diff
@@ -8,12 +8,24 @@
 import automata.base.config as global_config
 import automata.base.exceptions as exceptions
 from automata.base.utils import freeze_value
+from typing import Any, Mapping, NoReturn, AbstractSet, Tuple, Dict, Generator
 
+AutomatonStateT = Any
```

Comment:
@eliotwrobson So I remember when I was taking a stab on my own at adding types to this library, I was running into an issue where, because we were coercing most `__init__` input parameters to be immutable structures, the types of the `__init__` inputs wouldn't necessarily be the same as what the automata is storing.

For example, you could pass `List[Hashable]` to the `states` parameter on init, but that structure gets converted to something that looks more like `Set[Hashable]`, and the latter is what's actually stored on the instance. Another consequence of this was that I had to use `Hashable` in most places where `Any` would otherwise be used.

Do you have any thoughts on how to handle these complexities?

Reply:
@eliotwrobson Resolving this thread, as I've been using `AbstractSet` per your recommendation and have been pretty happy / haven't had any issues.